rtlib/src/ImageStitcher.cpp

```diff
@@ -0,0 +1,732 @@
+    for(int i = 0; i < points1.size(); i++){
+        // Implement search so it finds the index for the point
+        // Returns -1 if the points cannot be found
+        int index = search_(features, points1[i]);
+        if(index < 0){
+            // Check to see if this is the right way to do this
+            // Want to remove the points that are filtered out for points1
+            // Want to take out the match for the point that isn't there anymore
+            points1.erase(points1.begin() + i);
+            points2.erase(points2.begin() + i);
+            i--;
+        }
+    }
+}
+
+void ImageStitcher::reduce_img_points_(const double& work_scale, std::vector<std::pair<float, float> > &features1, std::vector<std::pair<float, float> > &features2){
+    for(int i = 0; i < features1.size(); i++){
+        features1[i].first = features1[i].first * work_scale;
+        features1[i].second = features1[i].second * work_scale;
+        features2[i].first = features1[i].first * work_scale;
+        features2[i].second = features2[i].second * work_scale;
+    }
+}
+
+int ImageStitcher::search_img_index_(int img_index) {
+    for(int i = 0; i < all_features_.size(); i++){
+        if(img_index == all_features_[i].img_idx){
+            return i;
+        }
+    }
+    return -1;
+}
+
+void ImageStitcher::insert_user_matches_(const LandmarkPair& ldm_pair){
+    // Need to have the images scaled for the work scale
+    double registr_resol_{0.6};
+    double work_scale_ = 1;
+    cv::Ptr<cv::FeatureDetector> orb = cv::ORB::create();
+    int src_size, dst_size;
+    // Populate the keypoints for the src image first
+    cv::detail::ImageFeatures img1_features;
+    int img1_idx = search_img_index_(ldm_pair.src_idx);
+    // Check if the src image is already in the features vector
+    if(img1_idx < 0){
+        // Creates a new feature and puts it into all_features_
+        img1_features.img_idx = ldm_pair.src_idx;
+        img1_features.img_size = cv::Size(imgs_[ldm_pair.src_idx].size().width, imgs_[ldm_pair.src_idx].size().height);
+        for(int i = 0; i < ldm_pair.src_ldms.size(); i++){
+            img1_features.keypoints.emplace_back(cv::KeyPoint(ldm_pair.src_ldms[i].first, ldm_pair.src_ldms[i].second, 31));
+        }
+
+        //May need to compute the descriptors
+
+        /*orb->detectAndCompute(smaller_imgs[0], cv::UMat(), feature_img1.keypoints, feature_img1.descriptors, true);
+        filterMatchedFeatures(feature_img1, features1, features2);*/
+        src_size = 0;
+        all_features_.push_back(img1_features);
+    }
+    else{
+        src_size = all_features_[img1_idx].keypoints.size();
+        // Appends the new landmarks
+        for(int i = 0; i < ldm_pair.src_ldms.size(); i++){
+            all_features_[img1_idx].keypoints.emplace_back(cv::KeyPoint(ldm_pair.src_ldms[i].first, ldm_pair.src_ldms[i].second, 31));
```

Comment:
`emplace_back` takes constructor arguments, not constructed objects.
```suggestion
            all_features_[img1_idx].keypoints.emplace_back(ldm_pair.src_ldms[i].first, ldm_pair.src_ldms[i].second, 31);
```